In [3]:
import sys
import os
import logging

# To‘g‘ri yo‘lni sys.path ga qo‘shish
sys.path.append(r"C:\Users\Rasulbek907\Desktop\Project_8\Source")

# Fayl nomi: Auto_Pipeline.py, shuning uchun import ham shunday bo'lishi kerak
from pipeline import DataLoader

# Log fayl manzili
log_path = r"C:\Users\Rasulbek907\Desktop\Project_8\Log\auto_pipeline.log"

# Log sozlamalari
logging.basicConfig(
    filename=log_path,
    filemode='a',
    format='%(asctime)s - %(levelname)s - %(message)s',
    level=logging.INFO
)

# Ma'lumot faylining manzili
data_path = r"C:\Users\Rasulbek907\Desktop\Project_8\Data\Feature_Selection\Filtered_Features.csv"

try:
    logging.info("Ma'lumot fayli yuklanmoqda...")
    df = DataLoader.load_csv(data_path)
    logging.info(f"Fayl muvaffaqiyatli yuklandi. Satrlar soni: {len(df)}, ustunlar soni: {len(df.columns)}")
except Exception as e:
    logging.error(f"Ma'lumot faylini yuklashda xatolik yuz berdi: {e}")

In [5]:
import sys
import os
import importlib

# 1. path qo‘shish
source_path = r'C:\Users\Rasulbek907\Desktop\amaliyot\Source'
if source_path not in sys.path:
    sys.path.append(source_path)

# 2. Import va reload
import pipeline
importlib.reload(pipeline)  # 🔄 MUHIM: o‘zgargan faylni yangilaydi

# 3. DataLoader ni olish
from pipeline import DataLoader

# 4. Foydalanish
data_path = r'C:\Users\Rasulbek907\Desktop\Project_8\Data\Feature_Selection\Filtered_Features.csv'
loader = DataLoader(data_path)

# 5. Endi ishlaydi!
df = loader.load_dataset()
print(df.head()) 

   Unnamed: 0     index  dateCrawled      name  vehicleType  model  \
0           0  0.000000     0.586679  0.339173     0.857143  0.472   
1           1  0.000003     0.585720  0.017608     0.428571  0.472   
2           2  0.000005     0.272956  0.387805     1.000000  0.476   
3           3  0.000008     0.378625  0.327975     0.571429  0.472   
4           4  0.000011     0.824117  0.730069     0.571429  0.412   

   monthOfRegistration  fuelType     brand  postalCode  lastSeen  \
0             0.000000  0.166667  0.974359    0.701176  0.978305   
1             0.416667  0.500000  0.025641    0.665989  0.975657   
2             0.666667  0.500000  0.358974    0.903792  0.891420   
3             0.500000  0.166667  0.974359    0.909796  0.247652   
4             0.583333  0.500000  0.794872    0.600115  0.940417   

   abtest_control  gearbox_automatik  notRepairedDamage_ja  kilometer_cont  
0             0.0                0.0                   0.0        0.005000  
1             0.

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 371528 entries, 0 to 371527
Data columns (total 15 columns):
 #   Column                Non-Null Count   Dtype  
---  ------                --------------   -----  
 0   Unnamed: 0            371528 non-null  int64  
 1   index                 371528 non-null  float64
 2   dateCrawled           371528 non-null  float64
 3   name                  371528 non-null  float64
 4   vehicleType           371528 non-null  float64
 5   model                 371528 non-null  float64
 6   monthOfRegistration   371528 non-null  float64
 7   fuelType              371528 non-null  float64
 8   brand                 371528 non-null  float64
 9   postalCode            371528 non-null  float64
 10  lastSeen              371528 non-null  float64
 11  abtest_control        371528 non-null  float64
 12  gearbox_automatik     371528 non-null  float64
 13  notRepairedDamage_ja  371528 non-null  float64
 14  kilometer_cont        371528 non-null  float64
dtype

In [7]:
import sys
import os
import importlib

source_path = r'C:\Users\Rasulbek907\Desktop\amaliyot\Source'
if source_path not in sys.path:
    sys.path.append(source_path)

import pipeline
importlib.reload(pipeline)  

from auto_pipeline import Auto_Pipeline

# Avto Pipeline RandomForestRegressor

In [9]:
from sklearn.ensemble import RandomForestRegressor

pipeline_rf = Auto_Pipeline(df=df, target="kilometer_cont", model=RandomForestRegressor(n_estimators=100))
pipeline_rf.fit()
y_pred = pipeline_rf.predict()

# R2 ballini olish
r2_score = pipeline_rf.score()
print("R2 score:", r2_score)

R2 score: 0.9303195244792568


In [10]:
import joblib
import os

model_path = r"C:\Users\Rasulbek907\Desktop\Project_8\Models\Pipeline_Models\RandomForestRegressor.joblib"
os.makedirs(os.path.dirname(model_path), exist_ok=True)

joblib.dump(pipeline_rf, model_path)
print(f"✅ Model muvaffaqiyatli saqlandi: {model_path}")  

✅ Model muvaffaqiyatli saqlandi: C:\Users\Rasulbek907\Desktop\Project_8\Models\Pipeline_Models\RandomForestRegressor.joblib


# Avto Pipeline DecisionTreeRegressor

In [12]:
from sklearn.tree import DecisionTreeRegressor

pipeline_dt = Auto_Pipeline(df=df, target="kilometer_cont", model=DecisionTreeRegressor())
pipeline_dt.fit()
y_pred = pipeline_dt.predict()

r2_score = pipeline_dt.score()
print("Decision Tree R2 score:", r2_score)

Decision Tree R2 score: 1.0


In [13]:
import joblib
import os

model_path = r"C:\Users\Rasulbek907\Desktop\Project_8\Models\Pipeline_Models\DecisionTreeRegressor.joblib"
os.makedirs(os.path.dirname(model_path), exist_ok=True)

joblib.dump(pipeline_dt, model_path)
print(f"✅ Model muvaffaqiyatli saqlandi: {model_path}")  

✅ Model muvaffaqiyatli saqlandi: C:\Users\Rasulbek907\Desktop\Project_8\Models\Pipeline_Models\DecisionTreeRegressor.joblib


# Avto Pipeline LinearRegression

In [16]:
from sklearn.linear_model import LinearRegression

pipeline_lr = Auto_Pipeline(df=df, target="kilometer_cont", model=LinearRegression())
pipeline_lr.fit()
y_pred = pipeline_lr.predict()

r2_score = pipeline_lr.score()
print("Linear Regression  R2 score:", r2_score)

Linear Regression  R2 score: 0.3524754557397375


In [17]:
import joblib
import os

model_path = r"C:\Users\Rasulbek907\Desktop\Project_8\Models\Pipeline_Models\LinearRegression.joblib"
os.makedirs(os.path.dirname(model_path), exist_ok=True)

joblib.dump(pipeline_lr, model_path)
print(f"✅ Model muvaffaqiyatli saqlandi: {model_path}")

✅ Model muvaffaqiyatli saqlandi: C:\Users\Rasulbek907\Desktop\Project_8\Models\Pipeline_Models\LinearRegression.joblib


# Natijalarni Taqqoslash 

# Auto Pipeline Regression Results

| Model                     | R2 Score |
|---------------------------|----------|
| **Decision Tree**         | <span style="color:green">1.0</span> |
| Random Forest             | <span style="color:orange">0.9303</span> |
| Linear Regression         | <span style="color:red">0.3525</span> |
